pilot/pkg/xds/endpoint_builder.go

```diff
@@ -286,6 +294,19 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 				ep.EnvoyEndpoint = buildEnvoyLbEndpoint(ep)
 			}
 			locLbEps.append(ep.EnvoyEndpoint, ep.TunnelAbility)
+
+			// compute the actual mTLS mode once policy applied, used during multinetwork filtering
+			k := ep.Labels.String()
+			// if the ep already had Disabled, leave it, otherwise resolve the policy
+			if tlsMode := envoytransportSocketMetadata(ep.EnvoyEndpoint, model.TLSModeLabelShortname); tlsMode != model.DisabledTLSModeLabel {
```

Comment:
if tlsMode is unset, doesn't that mean it has no sidecar and thus should not use mtls?

Reply:
I suppose its ambiguous from PeerAuthentication and depends on the DestinationRule.

Reply:
TLSMode on the IstioEndpoint gets set to Disabled if the k8s label is missing, but maybe we should be stricter and require it to be set.